src/page-multi-facility/MultiFacilityImpactDashboard.tsx

```diff
@@ -64,8 +64,23 @@ const MultiFacilityImpactDashboard: React.FC<Props> = ({
   useEffect(() => {
     async function fetchScenario() {
       const result = await baselineScenario?.data?.get();
+
+      let scenario = result?.data();
+
+      if (!scenario.hasOwnProperty("promoStatuses")) {
+        scenario = {
+          ...scenario,
+          promoStatuses: {
+            dataSharing: true,
+            dailyReports: true,
+            addFacilities: true,
+          },
+        };
+        await updateScenario(scenario);
```

Comment:
Let me know if placing this here makes sense!

Reply:
Would love to hear from @macfarlandian or @sprad on this one. The main thing was that we wanted to ensure we didn't have to manually migrate all existing scenarios to have this new `promoStatuses` object field, without which the app would break. Is there a better way to accomplish this that you're aware of?

Reply:
I am inclined to say that this should be somewhere in the `database` modules instead, so that the UI component just receives the data structure it requires and doesn't have to patch it.

Reply:
I was thinking that as well, currently we are getting the `scenario` here from the already fetched `baselineScenario` with `baselineScenario.data.get().data()`. Would you recommend adding this to the `getBaselineScenarioRef` function in the database module, or should I add another function that is `getBaselineScenario` that would then return the scenario directly instead of the `ref` and update it with the missing property?

Reply:
unless @sprad disagrees, I think a new function that consumes the `ref`, updates it as needed, and just returns the scenario object is the way to go. Good idea! I'm for moving all database knowledge out of the UI components

Reply:
@macfarlandian Ok! This is ready for another look.

Reply:
To specifically answer the question in this PR, I agree that this code belongs in the DB module, but not in the `getBaselineScenarioRef` method.

To comment more globally though, we will want to come up with some sort of data migration process or tooling.  Otherwise, we won't really ever be able to trust our data and our code will have to be "smart" enough to work with every version and permutation of the data model throughout time.  As the code base grows, this will become untenable.

You can already see this happening with this PR.  We used to store these promo flags directly on the scenario document, but now we're storing them in a separate `promoStatuses` object which makes sense.  But, because we didn't get rid of the old flags, we're having to do stuff like this:  https://github.com/Recidiviz/covid19-dashboard/blob/097d4eb94c442cc1f7819389706a4c942454e71e/src/page-multi-facility/ScenarioSidebar.tsx#L51-L57 It's going to be very confusing in the future if someone is looking at the data for a scenario and sees both  `dailyReports` and `promoStatuses.dailyReports`.   Which should they trust?  What if their values differ?

Additionally, we're technically still doing a migration here it's just that we're doing it in slowly in production application code whenever a user who doesn't meet these conditions logs in and views this page.  At the moment, we only have about ~30 scenarios in our DB.  Once all of the users are up to date with this new `promoStatuses` model this code will no longer serve a purpose.  And unless we go back and regularly check if everyone's documents have been updated, we'll have to leave this no-op code around indefinitely.

Reply:
This is the "this code" I'm referring to in the last paragraph:

https://github.com/Recidiviz/covid19-dashboard/blob/097d4eb94c442cc1f7819389706a4c942454e71e/src/database/index.tsx#L204-L213

And perhaps even the entire new `getBaselineScenario` function, unless we find a use case for it later.

Reply:
@sprad, I'll file an issue for managing database migrations to be completed at some point in the near to medium future. All of those points are well met!

However, I think you maybe misunderstood the code a tiny bit, or else I did. There is no `dailyReports` field on the scenario object, just a `promoStatuses` field that includes its own `dailyReports` field. The `dailyReports` that you are referring to, if I understand, is just on the client side. Let me know if I'm wrong on that.

Otherwise, for now, let's just focus on an approach that works without too much cruft. We'll remove any outdated fields from the data model, like the original attempt with the `showPromo` boolean, and try to have the code for ensuring we have default `promoStatuses` isolated to as small of a location as possible.

Reply:
@jessex - Unfortunately, we do already have both `dailyReports` and `dataSharing` fields on existing scenarios in our DB:

<img width="1263" alt="Screen Shot 2020-04-22 at 3 25 40 PM" src="https://user-images.githubusercontent.com/25503/80025040-fdc55c80-84ad-11ea-8d4e-54fa75a87809.png">

Additionally, we're still setting them when we create a new baseline scenario, which we will need to stop doing (cc @daschi ):

https://github.com/Recidiviz/covid19-dashboard/blob/097d4eb94c442cc1f7819389706a4c942454e71e/src/database/index.tsx#L233-L239

Note they are already set differently (opt-in vs. opt-out)

Reply:
Shit, sorry.  I am conflating two issues.  One is the actual toggle of whether or not we want data sharing or reporting.  The other is whether or not we want show promotion for these features.

Reply:
Yeah sorry for that confusion, but we're on the same page now. If you think it would be simpler, we can just change the names of the internal flag fields within `promoStatuses`.

Reply:
Should we wait until a migration process is ready for the data? Should I add a note above property check inside `getBaselineScenario` in the meantime?

Reply:
No, let's not wait until we have a migration process constructed. Let's just get this into a mergeable, working state for all users, and then we can manually remove the `showPromo` boolean fields we had added previously.

Reply:
@daschi - For now, no need to wait for a migration process.  I think adding a note above the property check in `getBaselineScenario` that references the issue @jessex is creating would suffice ✔️

Reply:
That issue lives here: #186